src/main/java/com/lmax/disruptor/MultiProducerSequencer.java

```diff
@@ -116,37 +116,24 @@ public long next(int n)
             throw new IllegalArgumentException("n must be > 0");
         }
 
-        long current;
-        long next;
-
-        do
-        {
-            current = cursor.get();
-            next = current + n;
-
-            long wrapPoint = next - bufferSize;
-            long cachedGatingSequence = gatingSequenceCache.get();
+        long current = cursor.getAndAdd(n);
 
-            if (wrapPoint > cachedGatingSequence || cachedGatingSequence > current)
-            {
-                long gatingSequence = Util.getMinimumSequence(gatingSequences, current);
-
-                if (wrapPoint > gatingSequence)
-                {
-                    LockSupport.parkNanos(1); // TODO, should we spin based on the wait strategy?
-                    continue;
-                }
+        long nextSequence = current + n;
```

Comment:
Aren't you adding `n` twice here, once on this line and once on 119 ?

Reply:
The call `long nextSequence = cursor.getAndAdd(n)` on line 119 will return the current value before adding n. So the sequence to return should be `current + n`

Reply:
👍